### ISSUE TITLE
Issue with dynamic imports

### ISSUE THREAD

**Issue by CodySchaaf** [bluetech/ng-annotate-patched#6]
The following code produces a parse error when compiling through webpack:

```
class Foo {
	constructor() {
		import('./module').then(foo => {
			console.log(foo);
		});
	}
}

new Foo();
```

ERROR in ./src/main.ts
Module build failed: error: couldn't process source due to parse error,Unexpected token (22:14)

I was able to find https://github.com/kesne/acorn-dynamic-import but it seems like you can't really make it optional. Any idea if this could be incorporated into this project, or would my best bet be forking it and adding this parser as the default.

**Comment by CodySchaaf** [comment]
Seems like it will be a bit more complicated than just plugging it in too. I gave it a shot and ran into:

```
ERROR in ./src/main.ts
Module build failed: TypeError: walk.base[(override || node.type)] is not a function
```

**Comment by jkepps** [comment]
Hey @CodySchaaf, any progress with this? I ran into the same issue.

**Comment by CodySchaaf** [comment]
@jkepps I have a fix ready https://github.com/kesne/acorn-dynamic-import/pull/10 but I'm waiting on a response from the repo owner. I could split the change into its own module `acorn-dynamic-import-walk` if I don't get a response. 

Right now I have my project pointing to a tmp branch https://github.com/CodySchaaf/ng-annotate-patched/tree/with-dynamic-import-support-tmp-my-version-of-acorn-fork , but I want to merge that into this repo since that temporary version is a little hacky while I wait for the other pr to be merged.

**Comment by jkepps** [comment]
Nice, thanks for the update.

I just tried using your tmp branch of ng-annotate-patched in my project, but I'm still getting the same build error from webpack:
```bash
ERROR in ./app/scripts/app.module.js
Module build failed: error: couldn't process source due to parse error,Unexpected token (69:10)
 @ ./app/scripts/index.js 1:0-23
```
line 69 is the first instance of a dynamic import.

here's my loader config:
```js
...
     {
        test: /\.js$/,
        use: {
          loader: 'ng-annotate-loader',
          options: {
            ngAnnotate: 'ng-annotate-patched',
            es6: true,
            explicitOnly: false
          }
        }
      },
...
```

and the particular commit i'm using in my package.json:
```json
"ng-annotate-patched": "github:CodySchaaf/ng-annotate-patched#79b1d89c2212f8195f3ca23a335590683cf29be3",
```

is there something else I'm missing?

**Comment by CodySchaaf** [comment]
Also add `dynamicImport: true` to the loader options:

```
...
     {
        test: /\.js$/,
        use: {
          loader: 'ng-annotate-loader',
          options: {
            ngAnnotate: 'ng-annotate-patched',
            es6: true,
            dynamicImport: true,
            explicitOnly: false
          }
        }
      },
...
```

See the added info to the readme https://github.com/CodySchaaf/ng-annotate-patched/commit/79b1d89c2212f8195f3ca23a335590683cf29be3.  I should probably make that more prominent, or maybe even the default depending on what the owner of this project wants.

**Comment by mariuszbeltowski** [comment]
@CodySchaaf @jkepps Could you please take a look at my problem? https://github.com/bluetech/ng-annotate-patched/issues/7

**Comment by bluetech** [comment]
Fixed by #8.
